### PR TITLE
Add `start_spoof` and `stop_spoof` cheatcodes

### LINKED ISSUE #1806: `start_spoof` cheatcode
We should be able to provide message signature from `get_tx_info` syscall in tests, in order to enable testing accounts

### PR DESCRIPTION
**Pull request by cptartur**
Closes #1806 

Related cairo-protostar compiler PR: https://github.com/MaksymilianDemitraszek/cairo-protostar/pull/3

### Checklist
- [x] Relevant issue is linked
- [ ] Docs updated/issue for docs created
- [x] Added relevant tests

Docs will be added in a separate PR. Issue: https://github.com/software-mansion/protostar/issues/2105

### CODE REVIEW

File: protostar/cheatable_starknet/controllers/transaction_info.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
+from dataclasses import dataclass
```
Comment (MaksymilianDemitraszek) [inline protostar/cheatable_starknet/controllers/transaction_info.py:1]: Test which would be beneficial here
- deploys contracts A and B
- spoofs contract B
- calls contract A, A calls B b checks if its spoofed

**Review by MaksymilianDemitraszek — APPROVED** [review]
LGTM and great job, please create a task for docs before merging

Can you @cptartur skim it briefly before merging it? To validate if everything is as its supposed to be

**Review by cptartur — COMMENTED** [review]
It looks fine interface-wise.

**Comment by abulenok** [comment]
@karol-bisztyga I added info about spoof cheatcodes to the linked issue.

File: protostar/cheatable_starknet/controllers/transaction_info.py (hunk 1 above)
Comment (abulenok) [reply protostar/cheatable_starknet/controllers/transaction_info.py:1]: Added

**Pull request merged**